### PR TITLE
KubeVirt prevent marshalling of resources both in instancetype and NodeSpec

### PR DESCRIPTION
**Pull request by hdurand0710**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes kubermatic/kubermatic#11847

**What type of PR is this?**

/kind bug

**Special notes for your reviewer**:
No backport needed.

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Comment by hdurand0710** [comment]
/hold 
to give @mfranczy some time to review too.

File: modules/api/pkg/api/v1/types.go [hunk 1]
```diff
@@ -2077,6 +2077,20 @@ func (spec *KubevirtNodeSpec) MarshalJSON() ([]byte, error) {
...
+	if spec.Instancetype != nil || len(spec.FlavorName) > 0 {
+		if len(spec.CPUs) > 0 {
+			doubleSpec = append(doubleSpec, "cpus")
+		}
+		if len(spec.Memory) > 0 {
+			doubleSpec = append(doubleSpec, "memory")
+		}
+	}
+	if len(doubleSpec) > 0 {
+		return []byte{}, fmt.Errorf("double definition  (instancetype/flavor and NodeSpec) of some parameter(s): %s", strings.Join(doubleSpec, ", "))
```
Comment (mfranczy) [inline modules/api/pkg/api/v1/types.go:2091]: I would maybe change the error message a little bit:
> cpu and memory at the same time cannot be set in template (instancetype/flavor) and node spec

**Review by mfranczy — CHANGES_REQUESTED** [review]
A small nit

**Review by mfranczy — APPROVED** [review]
/approve
/lgtm

**Pull request merged**